src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1704,18 +1712,35 @@ def update_vm(cmd, resource_group_name, vm_name, os_disk=None, disk_caching=None
         vm.security_profile.uefi_settings = UefiSettings(secure_boot_enabled=enable_secure_boot,
                                                          v_tpm_enabled=enable_vtpm)
 
-    if enable_proxy_agent is not None or proxy_agent_mode is not None:
+    if enable_proxy_agent is not None or wire_server_mode is not None or imds_mode is not None or \
+            wire_server_access_control_profile_reference_id is not None or \
+            imds_access_control_profile_reference_id is not None or key_incarnation_id is not None:
```

Comment:
This judgment condition seems too complex. Is there any way to simplify it or make it more readable?

Reply:
Done